### PR TITLE
Docs/tweaks

### PR DESCRIPTION
**Pull request by SignpostMarv**
as discussed in #1323, with a few additional maintenance tweaks.

### CODE REVIEW

File: docs/templated_annotations.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+ * @template T
+ * @psalm-param T $t
+ * @return T
+ */
+function mirror($t) {
+    return $t;
+}
+
+$a = 5;
+$b = mirror(5); // Psalm knows the result is an int
```
Comment (weirdan) [inline docs/templated_annotations.md]:
```suggestion
$b = mirror(5); // Psalm knows the result is an int

$c = "foo";
$d = mirror($c); // Psalm knows the result is string
```

Comment (SignpostMarv) [reply docs/templated_annotations.md]: can't quite tell from github's preview- is the suggestion to remove line 21?

Comment (weirdan) [reply docs/templated_annotations.md]: no, to add couple of more lines, showing the method return type being polymorphic based on argument type.

Comment (SignpostMarv) [reply docs/templated_annotations.md]: ah :D

**Pull request merged**